### ISSUE TITLE
PD not schedules down peers

### ISSUE THREAD

**Issue by BusyJay** [tikv/pd#4027]
## Bug Report
A TiKV is killed by OOM. PD schedules 

### What did you do?
Setup a cluster with 4 nodes. One TiKV is killed by OOM. And balance-region-scheduler is removed.

### What did you expect to see?
PD schedules all replicas from the down TiKV to other nodes.

### What did you see instead?
PD schedules most of replicas but still leave some.
![image](https://user-images.githubusercontent.com/1701473/130592632-e8220dcb-b13e-427c-906a-da3062b84256.png)

### What version of PD are you using (`pd-server -V`)?
```
Release Version: v5.1.1
Edition: Community
Git Commit Hash: 7cba1912b317a533e18b16ea2ba9a14ed2891129
Git Branch: heads/refs/tags/v5.1.1
UTC Build Time:  2021-07-23 01:16:18
```

**Comment by BusyJay** [comment]
After restarting PD server, it starts scheduling again, and down-peer reaches 0 at the end.

**Comment by HunDunDM** [comment]
fixed by #4078
